src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs

```diff
@@ -441,7 +441,10 @@ public IActionResult GetHistory([FromQuery] WalletHistoryRequest request)
                     var transactionItems = new List<TransactionItemModel>();
 
                     // Sorting the history items by descending dates. That includes received and sent dates.
-                    List<FlatHistory> items = accountHistory.History.OrderByDescending(o => o.Transaction.SpendingDetails?.CreationTime ?? o.Transaction.CreationTime).ToList();
+                    List<FlatHistory> items = accountHistory.History
+                                                            .OrderBy(o => (o.Transaction.SpendingDetails?.IsSpentConfirmed() ?? false) ? 1 : 0)
```

Comment:
Is this correct in the case where the received transaction is both unspent and confirmed? If I am understanding the LINQ correctly, since the transaction lacks spending details it will be placed in the unconfirmed result set, and will then be further ordered within that group by creation time descending. This seems incorrect, it should be in the result set with the other confirmed transactions.

Reply:
Good pickup, fixed it